bin/search-function-test-utilizing-llm.py

```diff
@@ -0,0 +1,105 @@
+import os
+import sys
+import json
+from datetime import date
+
+import ailab.db as db
+from ailab.models import openai
+import ailab.db.finesse as finesse
+
+from ailab.db.finesse.test_queries import get_random_chunk
+
+# Constants
+TEST_VERSION = date.today()
+REQUIRED_QUESTIONS = 1
+CHARACTER_LIMIT = 14383
+STORAGE_PATH = "/home/vscode/finesse-data-2/qna"
+
+
+def load_prompts_and_template(prompt_path):
+    """Loads prompts and template from provided path"""
+    system_prompt = finesse.load_prompt(prompt_path, "qna_system_prompt.txt")
+    user_prompt = finesse.load_prompt(prompt_path, "qna_user_prompt.txt")
+    json_template = finesse.load_json_template(prompt_path)
+
+    return system_prompt, user_prompt, json_template
+
+
+def construct_user_prompt(user_prompt, random_chunk_str, json_template):
+    """Constructs the user prompt using the user prompt, random chunk and json template"""
+    return (
+        f"{user_prompt}\n\nHere is the JSON containing the search:\n{random_chunk_str}"
+        f"\n\nAnd here is the JSON template:\n{json_template}"
+    )
+
+
+def generate_question(system_prompt, user_prompt, json_template, project_db):
+    """Generates a question and saves it to a file"""
+    average_tokens = 0
+    for i in range(REQUIRED_QUESTIONS):
+        with project_db.cursor() as cursor:
+            random_chunk = get_random_chunk(cursor)
+            if not random_chunk:
+                print("No chunk found in the database.")
+                sys.exit(1)  # exit the program if chunk is empty
+
+        constructed_user_prompt = construct_user_prompt(
+            user_prompt, str(random_chunk), json_template
+        )
+        total_length = len(system_prompt) + len(constructed_user_prompt)
+        average_tokens += total_length
```

Comment:
we sum the `total_length` before checking if it's below the `CHARACTER_LIMIT`. Make sure this is what we want.

Reply:
Indeed, the changes is done now.